### ISSUE TITLE
problem with endpoints

### ISSUE THREAD

**Issue by paulheu** [eiz/SynchronousAudioRouter#15]
I am having issues where I initially made some endpoints which seem to stick around after having removed them and added new ones. Possibly a registry issue, running WIN10 Creators update. 

I get things working by choosing the endpoints I see in the applications I use as well as desktop and then just trying the available ones in Reaper until I find a match.. So working yes, but obviouslynot how it's intended

**Comment by pkviet** [comment]
+1 same here. Additionally I have to open  Reaper as administrator in order to the SAR records or play peripherals to be available to reaper.

**Comment by paulheu** [comment]
It appears the issue is with how WIN10 now uses the trusted installer. The endpoint builder seems to not have full access to the registry entry and so can't remove or change it once in's set up

**Comment by LDEDRICK** [comment]
What I noticed is when you delete and remake an Endpoint it retains the name of the old endpoint on those channels.

If I make an endpoint called OBS, delete it then remake it as Microphone Out, Windows retains that it was called OBS originally. In sound devices it is still called OBS, I have to rename it to Microphone Out manually, this is probably related to the same thing.

**Comment by paulheu** [comment]
Yes, from what I could see it is because the 'Trusted installer' owns the registry entry and the software is not allowed to edit or remove it.

**Comment by eiz** [comment]
Investigating this now.

**Comment by ggiordantx** [comment]
I can't get SAR to create any endpoints.  All the dialogs seem to work but the endpoints never create. I get an error message "There was an error opening the audio hardware:  Error initializing ASIO buffers". This is from Reaper.  The ASIO device I am selecting for the hardware interface works in Reaper.  Any thoughts on what to look at.  I have removed and reinstalled.  Same result.

**Comment by paulheu** [comment]
If you are running latest official update for Windows10 this is due to the driver being unsigned. You will need to disable this requirement in UEFI settings and potentially set 'test mode' which you can revert once the driver is installed. There is quite a few sites that will explain how to do this, I do not have one handy atm so do a search for it.

If the driver for SAR shows as having an issue in Device Manager this is likely the issue you have.

**Comment by ggiordantx** [comment]
Thanks for the reply.  I don’t see any issues in Device Manager. In fact is shows the driver is signed.

[cid:image001.png@01D2DEB2.CF60AC90]

I get no errors or warnings on install.

But it can’t seem to access any of the other devices asio drivers.

[cid:image002.png@01D2DEB2.E32910F0]

Regards,
G^2

**Comment by paulheu** [comment]
MAke sure the ASIO driver is not in Safe mode as that will prevent using it

![image](https://user-images.githubusercontent.com/12208547/26836864-e973ee34-4ad3-11e7-9620-ed3874d9a780.png)

**Comment by ggiordantx** [comment]
This is not an option.  As far as I know the ASIO driver of the device is working fine.  I can connect to it from other applications.  It just won’t work in the audio router…

Thanks,
G^2

**Comment by paulheu** [comment]
And what ASIO driver would that be?

**Comment by ggiordantx** [comment]
Well, I have tried it with my Soundblaster, ASIOfofAll and VoicemeeterBanana. As the hardware interface and all produce same results and all can be used in other programs like Reaper or Jack.

Regards,
G^2

Glenn Giordano

On Jun 6, 2017, at 13:13, paulheu <notifications@github.com<mailto:notifications@github.com>> wrote:


And what ASIO driver would that be?

**Comment by kelan236** [comment]
Hey there.  I'm seeing the same behavior as paulheu mentioned in his original post.  I'm very new to this and I've been trying to mess around with it in the evenings with Reaper.  This issue has had me thoroughly confused about what I was doing.  Input/output names weren't matching up, endpoints didn't seem to be deleting.  I think I've fiddled with it so much adding/removing endpoints and applications trying to understand what it's doing that I have no idea where I am now.  I was just wondering if there have been any further developments on this issue?

Also, I'm not sure if this is related or not but it seems like I'm not able to save the correct endpoints to my programs.  For example, I currently have 3 playback endpoints.  Chrome Playback, which I created first, then VLC Playback, and then IE Playback.  I set the endpoint for Chrome successfully, but when I try to set the endpoint for VLC it seems to keep getting flipped over to the Chrome Playback endpoint.  Even if I set VLC to the VLC Playback endpoint in the Application Details window and click OK, and then immediately double click VLC to go back into the Application Details window, it has already flipped over to the Chrome Playback endpoint.  And the same with Internet Explorer, when I try to set it to IE Playback it keeps flipping over to either the VLC Playback or the Chrome Playback endpoint instead.  Would this weird behavior have anything to do with the things about registry and insufficient permissions and such that have been mentioned in this thread?

**Comment by Slawagn** [comment]
Windows 7, also have the issue from the first post

**Comment by 2L40K** [comment]
@kelan236 As I can see, there are no developments on what @LDEDRICK has explained.

But how do you set the Chrome and IE Playback endpoints?
Only VLC has options for that purpose, for Chrome and IE you need to use:
"Enable per-application audio routing rules"
And it's not possible to see inside Chrome or IE where their playback goes.

**Comment by 2L40K** [comment]
It appears that all other applications and drivers that I have tested have the same behavior - they don't delete their endpoints even after complete uninstallation.
